docs/src/grpc/reference.md

```diff
@@ -0,0 +1,681 @@
+  string type = 2;
+  // a set of headers tying the job to a workflow instance/task instance
+  JobHeaders jobHeaders = 3;
+  // a set of custom headers defined during modelling; currently returned as a serialized
+  // JSON document
+  string customHeaders = 4;
+  // the name of the worker which activated this job
+  string worker = 5;
+  // the amount of retries left to this job (should always be positive)
+  int32 retries = 6;
+  // when the job can be activated again, sent as a UNIX epoch timestamp
+  int64 deadline = 7;
+  // JSON document, computed at activation time, consisting of all visible variables to
+  // the task scope
+  string payload = 8;
+}
+
+message JobHeaders {
+  // the job's workflow instance key
+  int64 workflowInstanceKey = 1;
+  // the bpmn process ID of the job workflow definition
+  string bpmnProcessId = 2;
+  // the version of the job workflow definition
+  int32 workflowDefinitionVersion = 3;
+  // the key of the job workflow definition
+  int64 workflowKey = 4;
+  // the associated task element ID
+  string elementId = 5;
+  // the unique key identifying the associated task, unique within the scope of the
+  // workflow instance
+  int64 elementInstanceKey = 6;
+}
+```
+
+#### Errors
+
+##### GRPC_STATUS_INVALID_ARGUMENT
+
+Returned if:
+
+  - type is blank (empty string, null)
+  - worker is blank (empty string, null)
+  - timeout less than 1
+  - amount is less than 1
+
+
+### CancelWorkflowInstance RPC
+
+Cancels a running workflow instance
+
+#### Input: CancelWorkflowInstanceRequest
+
+```protobuf
+message CancelWorkflowInstanceRequest {
+  // the workflow instance key (as, for example, obtained from
+  // CreateWorkflowInstanceResponse)
+  int64 workflowInstanceKey = 1;
+}
+```
+
+#### Output: CancelWorkflowInstanceResponse
+
+```protobuf
+message CancelWorkflowInstanceResponse {
+}
+```
+
+#### Errors
+
+##### GRPC_STATUS_NOT_FOUND
+
+Returned if:
+
+  - no workflow instance exists with the given key. Note that since workflow instances
+    are removed once their are finished, it could mean the instance did exist at some point.
+
+
+### CompleteJob RPC
+
+Completes a job with the given payload, which allows completing the associated service task.
+
+#### Input: CompleteJobRequest
+
+```protobuf
+message CompleteJobRequest {
+  // the unique job identifier, as obtained from ActivateJobsResponse
+  int64 jobKey = 1;
+  // a JSON document representing the variables in the current task scope
+  string payload = 2;
+}
+```
+
+#### Output: CompleteJobResponse
+
+```protobuf
+message CompleteJobResponse {
+}
+```
+
+#### Errors
+
+##### GRPC_STATUS_NOT_FOUND
+
+Returned if:
+
+  - no job exists with the given job key. Note that since jobs are removed once completed,
+    it could be that this job did exist at some point.
+
+##### GRPC_STATUS_FAILED_PRECONDITION
+
+Returned if:
+
+  - the job was marked as failed. In that case, the related incident must be resolved before
```

Comment:
Or it is not activated?

Reply:
At the moment we allow a job to be completed even if not activated. We can discuss that.